.github/workflows/ci.yml

```diff
@@ -1,54 +1,18 @@
 name: ci
 
 on: [push, pull_request]
-
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  test:
     strategy:
+      fail-fast: false
       matrix:
-        ruby_version: [2.5.x, 2.6.x, 2.7.x]
-
+        os: [ubuntu-latest, macos-latest]
+        ruby: ['2.7', '3.0', '3.1', head, jruby, jruby-head, truffleruby, truffleruby-head]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Ruby
-      uses: actions/setup-ruby@v1
-      with:
-        ruby-version: ${{ matrix.ruby_version }}
-    - name: Calculate variable dynamic values
-      id: dynamic_values
-      run: |
-        echo "::set-output name=installed_ruby_version::$(ruby -e 'print RUBY_VERSION')"
-        echo "::set-output name=cacheTimeAnchor::$(ruby -e 'require %Q{date}; cacheExpirationSeconds = 60*60*24; print (Time.now.to_i / cacheExpirationSeconds)')"
-    - name: Download CodeClimate reporter
```

Comment:
Would be nice to keep the CodeClimate report

Reply:
Could you check now? I'm pretty novice with actions and CC